tuf/api/metadata.py

```diff
@@ -1765,42 +1973,73 @@ def to_dict(self) -> Dict[str, Any]:
             targets_dict["delegations"] = self.delegations.to_dict()
         return targets_dict
 
-    def add_key(self, role: str, key: Key) -> None:
+    def add_key(self, key: Key, role: Optional[str] = None) -> None:
         """Adds new signing key for delegated role ``role``.
 
+        If succinct_roles is used then the ``role`` argument is not required.
+
         Args:
-            role: Name of the role, for which ``key`` is added.
             key: Signing key to be added for ``role``.
+            role: Name of the role, for which ``key`` is added.
 
         Raises:
-            ValueError: If there are no delegated roles or if ``role`` is not
-                delegated by this Target.
+            ValueError: If the argument order is wrong or if there are no
+                delegated roles or if ``role`` is not delegated by this Target.
         """
-        if self.delegations is None or role not in self.delegations.roles:
+        # Verify that our users are not using the old argument order.
+        if isinstance(role, Key):
+            raise ValueError("Argument order has changed")
```

Comment:
nitpick: we shouldn't make guesses about how the error happened, we should just report what the situation is